src/features/course-main/course-main.test.tsx

```diff
@@ -73,3 +62,33 @@ describe('CourseMain', () => {
     expect(imageElement).toHaveAttribute('src', MOCKED_IMAGE_PATH);
   });
 });
+
+describe('CourseMain', () => {
+  it('renders the section with correct label depending on date', () => {
+    (useCourseByTitle as Mock).mockReturnValueOnce({
+      ...testCourse,
+      course: {
+        ...testCourse.course,
+        startDate: dayJS().format('D MMM, YYYY'),
+      },
+    });
+    render(<CourseMain courseName="Node.js course" type="Mentoring Program" />);
+    const labelElement = screen.getByText(Labels.AVAILABLE);
+    expect(labelElement).toBeVisible();
+  });
+});
+
+describe('CourseMain', () => {
+  it('renders the section with correct label depending on date', () => {
+    (useCourseByTitle as Mock).mockReturnValueOnce({
+      ...testCourse,
+      course: {
+        ...testCourse.course,
+        startDate: dayJS().add(1, 'month').format('D MMM, YYYY'),
+      },
+    });
+    render(<CourseMain courseName="Node.js course" type="Mentoring Program" />);
+    const labelElement = screen.getByText(Labels.UPCOMING);
+    expect(labelElement).toBeVisible();
+  });
+});
```

Comment:
Good job!

Reply:
To avoid unnecessary renders by beforeEach. But I removed last Describe